Realm/Realm/Handles/ThreadSafeReferenceHandle.cs

```diff
@@ -0,0 +1,33 @@
+﻿////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2017 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+namespace Realms
+{
+    internal class ThreadSafeReferenceHandle : RealmHandle
+    {
+        [Preserve]
+        public ThreadSafeReferenceHandle()
+        {
+        }
+
+        protected override unsafe void Unbind()
+        {
+            NativeCommon.delete_pointer((void*)handle);
```

Comment:
You can't use this deleter. You need a function that deletes a `ThreadSafeReferenceBase` pointer because it's the first type in the hierarchy to declare a virtual destructor.